node/src/effect.rs

```diff
@@ -254,11 +254,20 @@ pub trait EffectOptionExt {
     ///
     /// The function `f_some` is used to translate the returned value from an effect into an event,
     /// while the function `f_none` does the same for a returned `None`.
-    fn option<U, F, G>(self, f_some: F, f_none: G) -> Effects<U>
+    fn fold<U, F, G>(self, f_some: F, f_none: G) -> Effects<U>
     where
         F: FnOnce(Self::Value) -> U + 'static + Send,
         G: FnOnce() -> U + 'static + Send,
         U: 'static;
+
+    /// Finalizes a future returning an `Option` into two different effects.
+    ///
+    /// The function `f` is used to translate the returned value from an effect into an event,
+    /// In the case of `None`, empty vector of effects is returned.
+    fn map_some<U, F>(self, f: F) -> Effects<U>
```

Comment:
(Awesome! That's what I was missing when I tried to create finality signatures only if we're a validator! :tada:)